cratetorrent/src/peer.rs

```diff
@@ -232,8 +239,26 @@ impl PeerSession {
                 }
                 cmd = self.cmd_port.select_next_some() => {
                     match cmd {
+                        Command::Block { info, data }=> {
+                            match data {
+                                Ok(data) => {
+                                    peer_info!(self,
+                                        "Read from disk {}",
+                                        info
+                                    );
+                                    self.send_block(&mut sink, info, data).await?;
+                                }
+                                Err(e) => {
+                                    peer_info!(self,
+                                        "Error reading {}: {}",
+                                        info, e
+                                    );
+                                    todo!("handle disk read error");
```

Comment:
Peer session shouldn't be responsible for handling disk IO errors: that's very likely a torrent or engine level event.